src/files/models/AStar.java

```diff
@@ -1,4 +1,115 @@
 package files.models;
 
+import files.models.graph.Edge;
+import files.models.graph.Vertex;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.TreeSet;
+
 public class AStar {
+
+    private Graph graph;
+    int infinity = (int) Double.POSITIVE_INFINITY;
+
+    public static void main(String[] args) {
+        AStar aStar = new AStar();
+        aStar.startAStar();
+    }
+
+    public void startAStar() {
+        // Create graph
+        //graph = this.makeDijkstraGraph();
+        graph = this.makeSmallGraphA();
+        // A
+        Vertex startNode = graph.getVertex("A");
+        Vertex endNode = graph.getVertex("F");
+
+        // B
+        //Vertex startNode = graph.getVertex("J");
+        //Vertex endNode = graph.getVertex("F");
+
+        ArrayList<Vertex> result = aStar(startNode, endNode);
+
+    }
+
+    public Graph makeSmallGraphA() {
+        Graph myGraph= new Graph();
+        final Vertex A= myGraph.addVertex("A");
+        final Vertex B= myGraph.addVertex("B");
+        final Vertex C = myGraph.addVertex("C");
+        final Vertex D = myGraph.addVertex("D");
+        final Vertex E = myGraph.addVertex("E");
+
+        myGraph.newEdge(A,B, 5,  3);
+        myGraph.newEdge(A,C, 10,  3);
+        myGraph.newEdge(B,C, 3,  3);
+        myGraph.newEdge(B,D, 2,  3);
+        myGraph.newEdge(B,E, 9,  3);
+        myGraph.newEdge(C,B, 2,  3);
+        myGraph.newEdge(C,E, 1,  3);
+        myGraph.newEdge(D,E, 6,  3);
+        myGraph.newEdge(E,D, 4,  3);
+
+        return myGraph;
+    }
+
+    public ArrayList<Vertex> aStar(Vertex startNode, Vertex endNode) {
+
+        TreeSet<Vertex> openTreeSet = new TreeSet<>(Comparator.comparingLong(Vertex::getF));
+        ArrayList<Vertex> closedList = new ArrayList<>();
+
+        for (Vertex vertex : graph.getVertices()) {
+            vertex.setDistance(200);
```

Comment:
we can use infinity here. Will probably have to be reworked from 200 anyways, as our distance can be over 200 in our case.

Reply:
yea it was a quick test to make sure infinity wasn't causing problems again